packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
     final bool currentPackage = argResults['current-package'] && (argResults.wasParsed('current-package') || dartFiles.isEmpty);
     final bool flutterRepo = argResults['flutter-repo'] || inRepo(argResults.rest);
 
-    //TODO (pq): revisit package and directory defaults
-
-    if (currentDirectory  && !flutterRepo) {
-      // ./*.dart
-      final Directory currentDirectory = fs.directory('.');
-      bool foundOne = false;
-      for (FileSystemEntity entry in currentDirectory.listSync()) {
-        if (isDartFile(entry)) {
-          dartFiles.add(entry);
-          foundOne = true;
+    // Use dartanalyzer directly when possible
+    if (!flutterRepo) {
+      final String dartanalyzer = fs.path.join(
+          Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartanalyzer');
+      final List<String> cmd = <String>[dartanalyzer];
+      cmd.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (cmd.length < 2 || currentPackage) {
+        final Directory projDir = await projectDirFor(fs.currentDirectory.absolute);
+        if (projDir != null) {
+          cmd.add((projDir).path);
+        }
+      } else if (currentDirectory) {
+        cmd.add('.');
+      } else if (!cmd[1].startsWith(fs.currentDirectory.path)) {
```

Comment:
I don't understand what this is doing. Why does the path of the first command line argument matter?